### DISCUSSION TITLE
Disable transactions if only one MongoDB host is configured

### DISCUSSION THREAD

**Discussion by tanwald** [quarkusio/quarkus#26808] [Q&A]
Hi, I use a standalone MongoDB Container for local development and a replicaset with 3+ hosts in production. Now that transactions are supported in Panache for MongoDB I get issues on my local stack. Is there a way to disable the `@Transactional` support with environment variables or profiles? I guess I'd have to set the same properties as with Quarkus MongoDB dev services but I couldn't find them - if they are exposed at all... Or does the dev service spin up multiple host? 

The exception message I get in transactional contexts: "Transaction numbers are only allowed on a replica set member or mongos..."

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
This is not possible at the moment to globally disable transaction for MongoDB. Your use case seems legit, you can open an enhancement request (as a github issue) if you want.

Current devservices for MongoDB only starts a single instance, and there is no support for a replicaset inside a single container as far as I know so it may be difficult to implement. 
Internaly, we have a test resource that spins up a MongoDB replicaset, you can use it in your test but you will still need to have a running MongoDB replicaset for local development.
